### PR TITLE
FIX route-maker and router-middleware for search-perf

### PR DESCRIPTION
**Pull request by imalberto**
- store.getRoutes are now cached, no need to opt any further
- deprecated querystring routing feature and save a call to store.getAppConfig !
- tests for caching

### CODE REVIEW

**Comment by imalberto** [comment]
Commit 777b0f4 updated with feedback from @caridy

**Comment by caridy** [comment]
+1

File: lib/app/middleware/mojito-router.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ Router.prototype = {
     handle: function(store, globalLogger, RouteMakerClass) {
         logger = globalLogger;
 
-        var self = this;
+        var that = this;
```
Comment (drewfish) [inline lib/app/middleware/mojito-router.js]: The convention in the mojito codebase is to use `my` (instead of `that`).

**Comment by imalberto** [comment]
commit fea41a5 update from @drewfish feedback

**Comment by imalberto** [comment]
This PR has been replaced by https://github.com/yahoo/mojito/pull/553. Closing.

**Pull request closed without merging**